pydy/tests/test_joints.py

```diff
@@ -0,0 +1,155 @@
+        q1 = Symbol('q1')
+        self.joint.orient_child('Axis', [q1, self.parent.frame.x])
+
+        # TODO other similar frame.orient() tests can be done.
+        # leaving more tests to discuss its necessity with others.
+
+
+class TestPinJoint():
+    def setup(self):
+        self.parent = Body('parent')
+        self.child = Body('child')
+        self.pinjoint = PinJoint('pinjoint', self.parent, self.child)
+
+    def test_pinjoint_init(self):
+        # default values
+        assert self.pinjoint.axis1 == self.parent.frame.x
+        assert self.pinjoint.axis2 == self.child.frame.x
+        assert self.pinjoint.parent_joint_point == self.parent.masscenter
+        assert self.pinjoint.child_joint_point == self.child.masscenter
+        assert self.pinjoint.parent_joint_vector == self.parent.frame.x
+        assert self.pinjoint.child_joint_vector == self.child.frame.x
+
+        assert self.parent_joint_point.vel(self.parent.frame) == 0
+        assert self.child_joint_point.vel(self.child.frame) == 0
+
+    def test_pinjoint_parameters(self):
+        self.pinjoint = PinJoint('pinjoint', self.parent, self.child,
+                                 (1, 0, 0), (0, 1, 0), axis1='x', axis2='y')
+        assert self.pinjoint.axis1 == self.parent.frame.z
+        assert self.pinjoint.axis2 == self.child.frame.y
+        point1 = self.parent.masscenter.locatenew(self.parent.frame, self.parent.frame.z)
+        point2 = self.child.masscenter.locatenew(self.child.frame, self.child.frame.y)
+        assert self.pinjoint.parent_joint_point == point1
+        assert self.pinjoint.child_joint_piont == point2
+        assert self.pinjoint.parent_joint_vector == self.parent.frame.z
+        assert self.pinjoint.child_joint_vector == self.chidl.frame.y
+
+    def test_joint_joint_functions(self):
+
+        # part 1 assining parent-child relationship
+        self.pinjoint.set_parent_child_rel()
+        assert self.child.parent == self.parent
+        assert self.parent.child == self.child
+
+        # part 2 locating joint point in bodies
+        self.pinjoint._locate_joint_point()
+        child_masscenter = self.child.masscenter
+        parent_masscenter = self.parent.masscenter
+        parent_joint_point = parent_masscenter.locatenew(self.name + '_Point', self.parent_joint_vector)
+        child_joint_point = child_masscenter.locatenew(self.name + '_Point', self.child_joint_vector)
+        assert self.pinjoint.parent_joint_vector == parent_joint_point
+        assert self.pinjoint.child_joint_vector == child_joint_point
+
+        # part 3 aligning axis1 and axis2
+        self.join_frames()
+        # sample implementation
+        # self.child.frame.orient(self.parent.frame, 'Axis', [0, self.parent.frame.x])
+        assert cross(self.pinjoint.axis1, self.pinjoint.axis2) == self.child.frame.z
+        assert cross(self.pinjoint.axis2, self.pinjoint.axis1) == - self.parent.frame.z
+        self.align_axes()
```

Comment:
I would try an exmple joint here and make sure you know what needs to happen. I said that you "align the axes" when we talked, but it is more subtle than that. Aligning axes with the mechanics API, is more like creating the proper ref frame ref frame orientation and creating the right points. It worth doing this manually to make sure you understand the needed operations.